tools/imageio/image.hpp

```diff
@@ -834,11 +834,10 @@ class ImageT : public Image {
         return data;
     }
 
-    virtual std::vector<uint8_t> getUNORMPackedPadded(
-            uint32_t c0, uint32_t c0Pad, uint32_t c1, uint32_t c1Pad,
-            uint32_t c2, uint32_t c2Pad, uint32_t c3, uint32_t c3Pad) const override {
-        assert((c0 + c0Pad + c1 + c1Pad + c2 + c2Pad + c3 + c3Pad) % 8 == 0);
-        const auto targetPackBytes = (c0 + c0Pad + c1 + c1Pad + c2 + c2Pad + c3 + c3Pad) / 8;
+    virtual std::vector<uint8_t> getUNORMPacked(
```

Comment:
The padding (rather shifting due to sbit behavior) was only needed for these `12X4` and `10X6` formats, it just that it was previously handled in this function because this was a packer designed for packed types (although PACKn packed types). So instead I moved this sbit case to the generic per-component converter, because these types are per-component, just with some insignificant bits.

Reply:
Thanks. Newcomers won't be wondering what happened to the  `Padded` version so a comment is unnecessary.